modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionContext.java

```diff
@@ -310,6 +311,16 @@ public void execute(RunnableX task, Consumer<Throwable> onError) {
             } catch (Throwable e) {
                 onError.accept(e);
 
+                if (e instanceof IgniteException) {
+                    return;
+                }
+
+                if (e instanceof AssertionError) {
```

Comment:
still not clear to me ((( if we want to defense here from some calcite assertions probably we can implement it somehow more accurate ? What test will fail if i comment this part ? i can`t find it quick ((

Reply:
We known that Calcite can throw AssertionError. Such error doesn't mean that should fail handler work for the case - it just error for one concrete query which just will be failed for user, we shouldn't pass it to fail handler. In case it will be our own AssertionError - it's also a bug which should be logged to have ability to analyze it.